assets/javascript/apps/pipeline/nodes/widgets.tsx

```diff
@@ -846,8 +849,76 @@ export function HistoryTypeWidget(props: WidgetParams) {
         <small className="text-red-500">{historyNameError}</small>
       </div>
     </>
-  )
-    ;
+  );
+}
+
+export function HistoryModeWidget(props: WidgetParams) {
+  const options = getSelectOptions(props.schema);
+  const userMaxTokenLimit = concatenate(props.nodeParams["user_max_token_limit"]);
+  const maxHistoryLength = concatenate(props.nodeParams["max_history_length"]);
+  const initialHistoryMode = concatenate(props.nodeParams["history_mode"]);
+  const [historyMode, setHistoryMode] = useState(initialHistoryMode || "Summarize");
```

Comment:
This should be the value, not the label
```suggestion
  const [historyMode, setHistoryMode] = useState(initialHistoryMode || "summarize");
```

Reply:
done